app/components/WebsocketProvider.tsx

```diff
@@ -460,6 +462,15 @@ class WebsocketProvider extends React.Component<Props> {
       }
     );
 
+    this.socket.on("users.demote", async (event: PartialWithId<User>) => {
+      if (auth.user && event.id === auth.user.id) {
+        await Promise.all([
+          documents.fetchPage({ limit: Pagination.defaultLimit }),
+          collections.fetchPage({ limit: Pagination.defaultLimit }),
+        ]);
```

Comment:
Logically, we need updated policies regarding documents and collections for that user. I hope this is seems to be the right way to achieve that? Another tactic could be `policies.remove(document.id)` for all accessible documents but the similar doesn't seem to work well for collections.

Reply:
Lets do that for documents (remove), I'm working on a more generic system for loading relationships but policies aren't defined as a real relationship yet. For collections refetching the page makes sense but we need to remove all the previous collection policies still as there may be more than a page of collections loaded in memory.

Reply:
Still, the same problem remains if the user has a collection opened which is returned in next page and the permissions are revoked(collection policies are removed). The user would see the Not Found page. But yeah, we've somewhat delegated the problem, to later pages.

Reply:
Maybe we should introduce a `fetchAll` method that will paginate through until done

Reply:
Seems more reliable. Can give it a try.

Reply:
Done!